src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -990,6 +990,18 @@ async ValueTask LogDocumentChangesAsync(int? generation, CancellationToken cance
                     // project must support compilations since it supports EnC
                     Contract.ThrowIfNull(newCompilation);
 
+                    // The compiler only uses this predicate to determine if CS7101: "Member 'X' added during the current debug session
+                    // can only be accessed from within its declaring assembly 'Lib'" should be reported. 
+                    // Prior to .NET 8 Preview 4 the runtime failed to apply such edits.
+                    // This was fixed in Preview 4 along with support for generics. If we see a generic capability we can disable reporting
+                    // this compiler error. Otherwise, we leave the check as is in order to detect at least some runtime failures on .NET Framework.
+                    // Note that the analysis in the compiler detecting the circumstances under which the runtime fails
+                    // to apply the change has both false positives (flagged generic updates that shouldn't be flagged) and negatives
+                    // (didn't flag cases like https://github.com/dotnet/roslyn/issues/68293).
```

Comment:
Seems like this sort of thing is why we have capabilities. If this bit of logic is permanent, then I think it makes more sense to add a capability for this scenario specifically, that this code can check, and then in put the logic in the [`Parse` method](https://github.com/dotnet/roslyn/blob/main/src/Features/Core/Portable/EditAndContinue/EditAndContinueCapabilities.cs#L72) so that if it sees the `GenericAddMethodToExistingType` capability, it can also set the new capability. Hopefully the runtime can also report the capability at some point, so the parse method doesn't have any special logic.

But arguably that is just moving deck chairs on the titanic, so up to you.

Reply:
We could add a new capability. Not sure how we would call it though. `AddPublicMemberAndItsInvocationAtTheSameTime` is a bit mouthful. Since the issue was fixed at the same time as `GenericAddMethodToExistingType` and it doesn't apply to Mono, I'm leaning towards keeping it part of `GenericAddMethodToExistingType`.